build.cake

```diff
@@ -15,67 +18,90 @@ Task("Clean")
 
 Task("Restore")
     .Does(() => {
-        NuGetRestore("./src/Shouldly.sln");
+        DotNetCoreRestore("./src/Shouldly.sln", new DotNetCoreRestoreSettings{
+            Verbosity = DotNetCoreVerbosity.Minimal,
+        });
     });
 
 GitVersion versionInfo = null;
+DotNetCoreMSBuildSettings msBuildSettings = null;
+
 Task("Version")
     .Does(() => {
         GitVersion(new GitVersionSettings{
             UpdateAssemblyInfo = false,
             OutputType = GitVersionOutput.BuildServer
         });
         versionInfo = GitVersion(new GitVersionSettings{ OutputType = GitVersionOutput.Json });
-        // Update project.json
-        var updatedProjectJson = System.IO.File.ReadAllText(shouldlyProj)
-            .Replace("1.0.0-*", versionInfo.NuGetVersion);
-
-        System.IO.File.WriteAllText(shouldlyProj, updatedProjectJson);
+        
+        msBuildSettings = new DotNetCoreMSBuildSettings()
+                            .WithProperty("Version", versionInfo.NuGetVersion)
+                            .WithProperty("AssemblyVersion", versionInfo.AssemblySemVer)
+                            .WithProperty("FileVersion", versionInfo.NuGetVersion);
     });
 
 Task("Build")
     .IsDependentOn("Clean")
     .IsDependentOn("Version")
     .IsDependentOn("Restore")
     .Does(() => {
-        MSBuild("./src/Shouldly.sln");
+        DotNetCoreBuild("./src/Shouldly.sln", new DotNetCoreBuildSettings()
+        {
+            Configuration = configuration,
+            MSBuildSettings = msBuildSettings
+        });
     });
 
 Task("Test")
     .IsDependentOn("Build")
     .Does(() => {
-        DotNetCoreTest("./src/Shouldly.Tests/Shouldly.Tests.csproj");
+        DotNetCoreTool("./src/Shouldly.Tests/Shouldly.Tests.csproj", "xunit", "-configuration Debug");
+        try {
+            if (isAppVeyor)
+                AppVeyor.UploadTestResults("./src/Shouldly.Tests/report.xml", AppVeyorTestResultsType.XUnit);
+        } catch(Exception)
+        {
+
```

Comment:
Yeah this is odd, was just doing this to debug. So it's actually throwing here, there is a `EnsureSuccessStatusCode` in the cake code, however the upload seems to work in that the tests all show up.

I'd like to know what is happening here, I'll do some more investigating.